### PR TITLE
fixed flaky test TestLog/stateful_set_logs_with_all_pods by splitting expected output string

### LINKED ISSUE #125048: [Flaking Test] TestLog/stateful_set_logs_with_all_pods
### Which jobs are flaking?

https://prow.k8s.io/job-history/gs/ppc64le-kubernetes/logs/periodic-kubernetes-unit-test-ppc64le

### Which tests are flaking?

`TestLog/stateful_set_logs_with_all_pods` in `staging/src/k8s.io/kubectl/pkg/cmd/logs/logs_test.go`

### Since when has it been flaking?

https://github.com/kubernetes/kubernetes/pull/124732
This flaking test case was newly added as part of the above PR

### Testgrid link

https://testgrid.k8s.io/ibm-k8s-unit-tests-ppc64le#periodic-k8s-unit-tests-ppc64le

### Reason for failure (if possible)

```
{Failed;Failed;  === RUN   TestLog/stateful_set_logs_with_all_pods
    logs_test.go:557: stateful set logs with all pods: expected to contain "[pod/test-sts-0/test-container] test log content for pod test-sts-0\n[pod/test-sts-1/test-container] test log content for pod test-sts-1\n". Output: "[pod/test-sts-1/test-container] test log content for pod test-sts-1\n[pod/test-sts-0/test-container] test log content for pod test-sts-0\n"
    --- FAIL: TestLog/stateful_set_logs_with_all_pods (0.08s)
;=== RUN   TestLog
--- FAIL: TestLog (1.55s)
;}
```

### Anything else we need to know?
Flake was reproducible with stress tool:
[root@centos-8-small-9f46c kubernetes]# go version
go version go1.22.2 linux/ppc64le

[root@centos-8-small-9f46c kubernetes]# stress ./logs.test -test.run ^TestLog
```
/tmp/go-stress-20240522T102036-3748936792
--- FAIL: TestLog (0.35s)
    --- FAIL: TestLog/stateful_set_logs_with_all_pods (0.02s)
        logs_test.go:557: stateful set logs with all pods: expected to contain "[pod/test-sts-0/test-container] test log content for pod test-sts-0\n[pod/test-sts-1/test-container] test log content for pod test-sts-1\n". Output: "[pod/test-sts-1/test-container] test log content for pod test-sts-1\n[pod/test-sts-0/test-container] test log content for pod test-sts-0\n"
FAIL


ERROR: exit status 1


/tmp/go-stress-20240522T102036-1557253734
--- FAIL: TestLog (0.35s)
    --- FAIL: TestLog/stateful_set_logs_with_all_pods (0.01s)
        logs_test.go:557: stateful set logs with all pods: expected to contain "[pod/test-sts-0/test-container] test log content for pod test-sts-0\n[pod/test-sts-1/test-container] test log content for pod test-sts-1\n". Output: "[pod/test-sts-1/test-container] test log content for pod test-sts-1\n[pod/test-sts-0/test-container] test log content for pod test-sts-0\n"
FAIL
```

### Relevant SIG(s)

/sig testing

### PR DESCRIPTION
**Pull request by srivastav-abhishek**
#### What type of PR is this?
/kind flake

#### Which issue(s) this PR fixes:

Fixes #125048

#### Special notes for your reviewer:
Fixed test ```TestLog/stateful_set_logs_with_all_pods``` by splitting expected o/p string to accommodate  situations where in Pod1 logs are present before Pod0 in buffer. 
#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/kubectl/pkg/cmd/logs/logs_test.go [hunk 1]
```diff
@@ -117,7 +117,7 @@ func TestLog(t *testing.T) {
 				o.Prefix = true
 				return o
 			},
-			expectedOutSubstrings: []string{"[pod/test-sts-0/test-container] test log content for pod test-sts-0\n[pod/test-sts-1/test-container] test log content for pod test-sts-1\n"},
+			expectedOutSubstrings: []string{"[pod/test-sts-0/test-container] test log content for pod test-sts-0\n", "[pod/test-sts-1/test-container] test log content for pod test-sts-1\n"},
```
Comment (eddiezane) [inline staging/src/k8s.io/kubectl/pkg/cmd/logs/logs_test.go:120]:
```suggestion
			expectedOutSubstrings: []string{
				"[pod/test-sts-0/test-container] test log content for pod test-sts-0\n",
				"[pod/test-sts-1/test-container] test log content for pod test-sts-1\n",
			},
```

**Review by eddiezane — CHANGES_REQUESTED** [review]
Thanks for the fix! Clean up request then lgtm.

/approve

File: staging/src/k8s.io/kubectl/pkg/cmd/logs/logs_test.go (hunk 1 above)
Comment (srivastav-abhishek) [reply staging/src/k8s.io/kubectl/pkg/cmd/logs/logs_test.go]: Addressed. Thanks !

**Review by eddiezane — APPROVED** [review]
Thank you!

/lgtm

**Pull request merged**